### ISSUE TITLE
Using PopUP as a two level menu and lead to memory issues, Terminated due to memory issue

### ISSUE THREAD

**Issue by kentaccn** [react-native-community/react-native-modal#162]
### Hi, I found there is abnormal memoery usage when using react-native-modal as a two level menu

normally the pop up contain a View component, i make it into a list parent menu click 
 TouchableWithoutFeedback, when it is click, it will turn into another flatlist let use to press the item, the render from list parent menu to flat list view is controled by style(display:none and flex) with state (true and false).

However, i found the memory rapidly increase after the click in list parent menu, when i go back to the list parent menu from the flat list view, the memory keep increasing and wont stop, the app use all memory after 2.5 style changes.

However, just one level popup is fine.

Do anyone have suggestions on this case, i have thought of componentWillUnmount() but it seem may not work on this case.  Ofcoz it is good to use the whole page redirect than popup UI design but it seem react-native-modal missing memory leak control solution in this special scenario.

Please correct me if i am wrong and thanks again to all contributors making this great app

dependencies:

        "color": "1.0.3",
        "flag-icon-css": "^3.0.0",
        "fontawesome": "^4.7.2",
        "jwt-decode": "^2.2.0",
        "lodash": "4.13.1",
        "moment": "2.13.0",
        "prop-types": "^15.6.0",
        "react": "16.0.0",
        "react-native": "^0.51.0",
        "react-native-modal": "^5.0.0",
        "react-native-modal-dropdown": "^0.6.0",
        "react-native-modalbox": "^1.4.2",
        "react-native-popup-dialog": "^0.10.46",
        "react-native-scrollable-tab-view": "^0.8.0",
        "react-native-search-box": "0.0.13",
        "react-native-svg": "^6.3.1",
        "react-native-svg-uri": "^1.2.3",
        "react-native-tab-navigator": "^0.3.4",
        "react-native-tabs": "^1.0.9",
        "react-native-vector-icons": "^4.6.0",
        "react-native-video": "^2.0.0",
        "react-navigation": "1.0.0-beta.19",
        "react-redux": "^5.0.6",

**Comment by mmazzarolo** [comment]
Hi!
Could you please reproduce the issue in a simple project/repo?

**Comment by mmazzarolo** [comment]
> Memory leak still happened as the flatlist
Memory leaks with lists are pretty common. Check you're not instantiating new objects inside loops/render(). Also, keep in mind that `console.log()` statement must be removed from the release mode otherwise they will fail and eat up memory.

> it is strange that on the emulator, it only eat up ~90MB memory but 500MB on real device. interesting

Unfortunately this is the norm with react-native. When using a simulator AND when running the device in debugging mode you will have way more memory than on release. 
Long story short: when checking your app perfomance always run on a real device in release mode.

**Comment by mmazzarolo** [comment]
Closing for now, feel free to create a new issue when needed!
